bioimageio/core/build_spec/build_spec.py

```diff
@@ -0,0 +1,447 @@
+import datetime
+import hashlib
+import os
+from pathlib import Path
+from typing import Any, Dict, List, Optional, Union
+
+import numpy as np
+from ruamel.yaml import YAML
+
+import bioimageio.spec as spec
+import bioimageio.spec.model as model_spec
+
+try:
+    from typing import get_args
+except ImportError:
+    from typing_extensions import get_args  # type: ignore
+
+#
+# utility functions to build the spec from python
+#
+
+
+def _get_local_path(uri, root=None):
```

Comment:
this could actually be substituted with `bioimageio.spec.shared.utils.resolve_uri`